Lib/asyncio/proactor_events.py

```diff
@@ -227,18 +228,45 @@ def _loop_reading__on_eof(self):
         if not keep_open:
             self.close()
 
-    def _loop_reading(self, fut=None):
-        self._loop_reading_cb(fut)
-
-    def _loop_reading__data_received(self, fut):
-        if (fut is not None and
-                getattr(fut, '__asyncio_cancelled_on_pause__', False)):
+    def _data_received(self, data):
+        if self._paused:
+            # don't call any protocol method while reading is paused
+            assert self._pending_data is None
+            self._pending_data = data
             return
 
-        if self._paused:
-            self._reschedule_on_resume = True
+        if not data:
+            self._eof_received()
             return
 
+        if isinstance(self._protocol, protocols.BufferedProtocol):
+            nbytes = len(data)
+            if nbytes:
+                try:
+                    buf = self._protocol.get_buffer(-1)
```

Comment:
You should pass `nbytes` to `get_buffer()`.  Also, `get_buffer()` can return a smaller buffer than requested. So you better should use `sslproto._feed_data_to_bufferred_proto` helper to make sure that all of the received data is passed to the `BufferedProtocol`.

Reply:
> you better should use sslproto._feed_data_to_bufferred_proto helper

Ok, done.